crates/iota-sdk-ffi/src/types/transaction/mod.rs

```diff
@@ -18,68 +27,62 @@ use crate::types::{
 ///
 /// transaction-v1 = transaction-kind address gas-payment transaction-expiration
 /// ```
-#[derive(Clone, Debug, derive_more::From, uniffi::Object)]
-pub struct Transaction(pub iota_types::Transaction);
-
-#[uniffi::export]
-impl Transaction {
-    #[uniffi::constructor]
-    pub fn new(
-        kind: &TransactionKind,
-        sender: &Address,
-        gas_payment: GasPayment,
-        expiration: &TransactionExpiration,
-    ) -> Self {
-        Self(iota_types::Transaction {
-            kind: kind.0.clone(),
-            sender: **sender,
-            gas_payment: gas_payment.into(),
-            expiration: expiration.0,
-        })
-    }
-
-    pub fn kind(&self) -> TransactionKind {
-        self.0.kind.clone().into()
-    }
-
-    pub fn sender(&self) -> Address {
-        self.0.sender.into()
-    }
+#[derive(Clone, Debug, uniffi::Record)]
+pub struct Transaction {
```

Comment:
Not sure about changing this to a record, I can imagine we want to have something like `from/to _base64()` for it, as this conversion happens often for signers
Probably same for `SignedTransaction` because posting it is usually also done with the bas64 encoded version
Or do you think we should have individual conversion functions for these that accept a record?